chore(ui-kit):Uploder component

jatinrise opened the pull request:
Hello @alebg,
I have incorporated all the changes as per our discussion.

Here are the updates I’ve made:

1. I’ve created a `generic` variant that accepts all file types and automatically displays a thumbnail based on the file type.
2. I’ve implemented the `AbortController` Web API to cancel the fetch request if the user clicks the cancel button.
3. I’ve added an id field to the `FileUploadRequestSchema` on the client side, which helps track the upload status and enables handling the onCancel functionality.

Hello @maany, please review my code and let me know if the approach I’ve used for aborting ongoing requests with `AbortController` is appropriate.

carlapbg commented on packages/ui-kit/lib/components/drag-and-drop-uploader/file-preview.tsx:
```diff
@@ -29,55 +28,95 @@ interface FilePreviewProps extends isLocalAware {
+        if (fileCategory === 'image') {
+            return (
+                <img
+                    src={(uploadResponse as fileMetadata.TFileMetadata & { thumbnailUrl?: string }).thumbnailUrl || ''}
```
Hi! In those cases, I think we should show an icon so that it doesn't look visually broken. For example, we could use the "image" icon that's in the Figma file: https://www.figma.com/design/8KEwRuOoD5IgxTtFAtLlyS/Just_Do_Ad-1.2?node-id=5373-120981&t=tk8PO9ZWLjrs6bbo-4

jatinrise commented on packages/ui-kit/lib/components/drag-and-drop-uploader/uploader.tsx:
```diff
@@ -183,70 +185,90 @@ export const Uploader: React.FC<UploaderProps> = (props) => {
+          return err;
+        }
+      } finally {
+        abortControllers.current.delete(tempId);
```
@alebg  yes, this is main reason to use id from client.

jatinrise commented:
Hello @alebg, I have incorporated all the inline comments.

1. The uploader component now handles all the simulated data internally.
2. Added a fallback icon in case the thumbnail fails to load.
3. The `document` preview now supports all file types except image and video.

Please review my changes.

maany reviewed (commented):
I have just reviewed the handling of AbortSignals to cancel File Uploads. It looks good to me to issue a client side abort. I am not 100% sure if MinIO will rollback the upload from a client side abort or publish an incomplete binary. 

@alebg as a follow-up, could you create a task for us to verify this later?

For the purpose of the upload client, the implementation looks alright. Nice work @jatinrise

alebg reviewed (approved):
Thanks a lot @jatinrise, great job! All remarks are covered
